src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -497,36 +520,88 @@ public PointSetTimes eval (
             }
-            int t = Math.min(t1, t2);
-            if (t != Integer.MAX_VALUE) {
-                if (t < distanceToPoint.get(p)) {
-                    distanceToPoint.put(p, t);
+
+            if (cost != Integer.MAX_VALUE) {
+                if (cost < costToPoint.get(p)) { // When is this false?
+                    costToPoint.put(p, cost);
                 }
             }
             return true; // Continue iteration.
         });
-        if (distanceToPoint.size() == 0) {
+        if (costToPoint.size() == 0) {
             return null;
         }
         // Convert a packed array of pairs.
         // TODO don't put in a list and convert to array, just make an array.
-        TIntList packed = new TIntArrayList(distanceToPoint.size() * 2);
-        distanceToPoint.forEachEntry((point, distance) -> {
+        TIntList packed = new TIntArrayList(costToPoint.size() * 2);
+        costToPoint.forEachEntry((point, distance) -> {
             packed.add(point);
             packed.add(distance);
             return true; // Continue iteration.
         });
         return packed.toArray();
     }
 
+    /**
+     * Given accumulated distances to vertices, add the remaining distance needed to reach a point that is linked to
+     * an edge. This remaining distance consists of a distance along part of the edge (to the "split point"), plus a
+     * perpendicular distance from the split point to the final point.
+     *
+     * @return minimum distance needed to reach point, or Integer.MAX_VALUE if point is not reachable.
+     */
+    private int distanceToPoint(TIntIntMap distancesToVertices, Edge edge, int pointIndex) {
+        int distance0 = Integer.MAX_VALUE;
+        int distance1 = Integer.MAX_VALUE;
+        // TODO this is not strictly correct when there are turn restrictions onto the edge this is linked to
+        if (distancesToVertices.containsKey(edge.getFromVertex())) {
+            distance0 = distancesToVertices.get(edge.getFromVertex()) + distances0_mm[pointIndex] + distancesToEdge_mm[pointIndex];
+        }
+        if (distancesToVertices.containsKey(edge.getToVertex())) {
+            distance1 = distancesToVertices.get(edge.getToVertex()) + distances1_mm[pointIndex] + distancesToEdge_mm[pointIndex];
+        }
+
+        return Math.min(handleOverflow(distance0), handleOverflow(distance1));
+    }
+
+    /**
+     * Given accumulated time to reach vertices, add the remaining time needed to reach a point that is linked to
+     * an edge. This remaining time consists of a time traversing part of the edge (at onStreetSpeed), plus a
+     * time needed to travel the perpendicular distance (at offStreetSpeed) from the split point to the final point.
+     *
+     * @return minimum time needed to reach point, or Integer.MAX_VALUE if point is not reachable.
+     */
+    private int timeToPoint(int time0, int time1, int onStreetSpeed, int offStreetSpeed, int pointIndex) {
```

Comment:
The Javadoc should ideally say what time0 and time1 are, and what the units of speed are.